node/src/neighborhood/gossip_acceptor.rs

```diff
@@ -2437,11 +2513,14 @@ mod tests {
             ),
             handle_result
         );
-        let expected_introducer = NodeRecord::from(&agrs[0]);
-        assert_eq!(
-            Some(&expected_introducer),
-            dest_db.node_by_key(&agrs[0].inner.public_key)
-        );
+        let mut expected_introducer = NodeRecord::from(&agrs[0]);
+        let actual_introducer = dest_db.node_by_key(&agrs[0].inner.public_key).unwrap();
+        let expected_last_update = expected_introducer.metadata.last_update;
+        expected_introducer.metadata.last_update = actual_introducer.metadata.last_update;
+        assert_eq!(&expected_introducer, actual_introducer);
+        assert! (((expected_last_update as i32) - (actual_introducer.metadata.last_update as i32)).abs() < 10,
```

Comment:
First, why don't we know the exact number? This is not a multinode test or something, here we don't chaotically send Gossip around. I'd expect it predictable so I'm surprised by the loose assertion. However, I didn't think too seriously about how this value is formed. 

Also, if we need to keep this assertion so complex, removing the parenthesis around the recast last update values would make it leaner and more readable, I think. Rust doesn't require the parenthesis. It knows it has to cast values as early as possible.  

`((expected_last_update as i32) - (actual_introducer.metadata.last_update as i32)).abs()` could be 
`(expected_last_update as i32 - actual_introducer.metadata.last_update as i32).abs()`

Reply:
They're timestamps. They should be very close, but because they're stamped at different places in the code, we can't guarantee that they'll be exactly the same. We also can't guarantee they'll be at most one millisecond apart, both because Actions VMs are slow and because the cheap low-resolution timers used by most OSes have a granularity greater than 1ms. (The granularity of the old 4.77MHz IBM PC's timer was 55.6ms, or 18 ticks per second; I still remember that.) The assumption here is that the system's timer granularity will be 10ms or less, and that not more than one tick transition will elapse during the test.

And I think the extra parentheses make it easier to read, not harder. However, I'll take 'em out for you.